benches/store.rs

```diff
@@ -0,0 +1,353 @@
+
+    for size in BATCH_SIZES {
+        let leaves = &random_data[..size];
+
+        let mtree_leaves: Vec<Word> = leaves.iter().map(|v| v.into()).collect();
+        let mtree = MerkleTree::new(mtree_leaves.clone()).unwrap();
+        let store = MerkleStore::new().with_merkle_tree(mtree_leaves).unwrap();
+        let depth = mtree.depth();
+        let root = mtree.root();
+
+        group.bench_function(BenchmarkId::new("MerkleTree", size), |b| {
+            b.iter_batched(
+                || {
+                    let sizeu64 = u64::try_from(size).unwrap();
+                    (rand_value::<u64>() % sizeu64) + sizeu64
+                },
+                |value| black_box(mtree.get_node(NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+
+        group.bench_function(BenchmarkId::new("MerkleStore", size), |b| {
+            b.iter_batched(
+                || {
+                    let sizeu64 = u64::try_from(size).unwrap();
+                    (rand_value::<u64>() % sizeu64) + sizeu64
+                },
+                |value| black_box(store.get_node(root, NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+    }
+}
+
+/// Benchmarks getting a node on SMT and MerkleStore backends.
+fn get_leaf_simplesmt(c: &mut Criterion) {
+    let mut group = c.benchmark_group("get_leaf_simplesmt");
+
+    let random_data_size = BATCH_SIZES.into_iter().max().unwrap();
+    let random_data: Vec<RpoDigest> = (0..random_data_size)
+        .map(|_| rand_array::<Felt, 4>().into())
+        .collect();
+
+    for size in BATCH_SIZES {
+        let leaves = &random_data[..size];
+
+        let smt_leaves = leaves
+            .iter()
+            .enumerate()
+            .map(|(c, v)| (c.try_into().unwrap(), v.into()))
+            .collect::<Vec<(u64, Word)>>();
+        let smt = SimpleSmt::new(63)
+            .unwrap()
+            .with_leaves(smt_leaves.clone())
+            .unwrap();
+        let store = MerkleStore::new()
+            .with_sparse_merkle_tree(smt_leaves)
+            .unwrap();
+        let depth = smt.depth();
+        let root = smt.root();
+
+        group.bench_function(BenchmarkId::new("SimpleSmt", size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(smt.get_node(&NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+
+        group.bench_function(BenchmarkId::new("MerkleStore", size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(store.get_node(root, NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+    }
+}
+
+/// Benchmarks getting an invalid index on the SMT and MerkleStore backends.
+fn get_leaf_simplesmt_invalid(c: &mut Criterion) {
```

Comment:
I think what we want to benchmark here is not invalid access, but accessing an empty node. Since we are building an SMT with relatively few elements (i.e., at most 1024) almost any random index should yield an empty node - so, we probably don't need to iterate over batch size here.

Reply:
added